packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -86,6 +92,22 @@ export class RedirectsMiddleware extends MiddlewareBase {
         return res || NextResponse.next();
       }
 
+      // Find context site language and replace token
+      if (REGEXP_CONTEXT_SITE_LANG.test(existsRedirect.target)) {
+        const siteInfo = this.config.sites?.find(
+          (siteItem: SiteInfo) => siteItem.name === site?.name
+        );
+
+        if (!siteInfo) {
+          throw new Error(`Could not resolve context site language for site ${site.name}`);
+        }
+
+        existsRedirect.target = existsRedirect.target.replace(
+          REGEXP_CONTEXT_SITE_LANG,
+          siteInfo?.language
+        );
+      }
+
```

Comment:
Unrelated, we may want to consider adding the same [try/catch in `getHandler` setup](https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss-nextjs/src/middleware/multisite-middleware.ts#L28-L38) we have for error handling in other middleware. Noticed we didn't apply that to this middleware.